controller/network/path_test.go

```diff
@@ -302,3 +303,79 @@ func TestShortestPath(t *testing.T) {
 		9 + 20 + 21 + 4 // link2 cost and src and dest latency plus dest router cost
 	req.Equal(int64(expected), cost)
 }
```

Comment:
Can you add a test for if the non-traversable router is the only router connecting the initiating and terminating routers and make sure that circuit creation fails in that case? You'll likely have to change the path selection algorithm

Reply:
I can! One clarification though. I didn't have it completely not connect as to not lock connections. Was that not something I need to worry about?

Reply:
Turns out I had unintentionally had it work that way anyway. Completely missed the error throwing on costs over maxint32. Added a unit test to cover this case.

Reply:
:+1: 

Can you add a couple more test cases?

Both of these should work, as they're on the ends, not in the middle
Untraversable <-> Untraversable 
Untraversable <-> Traversable <-> Untraversable

Reply:
Added